natlas-server/app/api/routes.py

```diff
@@ -91,7 +91,7 @@ def submit():
 			response_body = json.dumps({"status": status_code, "message": "XML had too many hosts in it", "retry": False})
 
 		# If it's not an acceptable target, tell the agent it's out of scope
-		elif not current_app.ScopeManager.isAcceptableTarget(nmap.hosts[0].address):
+		elif len(nmap.hosts) == 1 and current_app.ScopeManager.isAcceptableTarget(nmap.hosts[0].address):
```

Comment:
You removed the not. Is that expected?

I read this as..

If length of nmap.hosts == 1 and target is acceptable: return 400

Also, Should this be >= 1?

Reply:
Good catch, the not is required but currently the hosts count must be exactly 1. We don't have support for batch scan importing yet (but it's on it's way, there are just some caveats that are being worked out).